internal/pkg/services/load-balancer/utils/utils_test.go

```diff
@@ -793,3 +841,314 @@ func TestGetTargetName(t *testing.T) {
+			usedCredentials: []loadbalancer.CredentialsResponse{
+				{
+					DisplayName:    utils.Ptr("credentials-1"),
+					CredentialsRef: utils.Ptr("credentials-ref-1"),
+					Username:       utils.Ptr("user-1"),
+				},
+			},
+			isValid:        true,
+			expectedOutput: nil,
+		},
+		{
+			description:     "no used credentials, no credentials",
+			allCredentials:  []loadbalancer.CredentialsResponse{},
+			usedCredentials: nil,
+			isValid:         true,
+			expectedOutput:  nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			output := GetUnusedObsCredentials(tt.usedCredentials, tt.allCredentials)
+
+			diff := cmp.Diff(output, tt.expectedOutput)
+			if diff != "" {
+				t.Fatalf("Data does not match: %s", diff)
+			}
+		})
+	}
+}
+
+func TestFilterCredentials(t *testing.T) {
+	tests := []struct {
+		description            string
+		filterOp               int
+		allCredentials         []loadbalancer.CredentialsResponse
+		listLoadBalancersResp  *loadbalancer.ListLoadBalancersResponse
+		listLoadBalancersFails bool
+		expectedCredentials    []loadbalancer.CredentialsResponse
+		isValid                bool
+	}{
+		{
+			description:         "unfiltered credentials",
+			filterOp:            OP_FILTER_NOP,
+			allCredentials:      fixtureCredentials(),
+			expectedCredentials: fixtureCredentials(),
+			isValid:             true,
+		},
+		{
+			description:    "used credentials",
+			filterOp:       OP_FILTER_USED,
+			allCredentials: fixtureCredentials(),
+			listLoadBalancersResp: &loadbalancer.ListLoadBalancersResponse{
+				LoadBalancers: &[]loadbalancer.LoadBalancer{
+					*fixtureLoadBalancer(),
+				},
+			},
+			expectedCredentials: []loadbalancer.CredentialsResponse{
+				{
+					CredentialsRef: utils.Ptr("credentials-ref-1"),
+					DisplayName:    utils.Ptr("credentials-1"),
+					Username:       utils.Ptr("user-1"),
+				},
+				{
+					CredentialsRef: utils.Ptr("credentials-ref-2"),
+					DisplayName:    utils.Ptr("credentials-2"),
+					Username:       utils.Ptr("user-2"),
+				},
+			},
+			isValid: true,
+		},
+		{
+			description:    "unused credentials",
+			filterOp:       OP_FILTER_UNUSED,
+			allCredentials: fixtureCredentials(),
+			listLoadBalancersResp: &loadbalancer.ListLoadBalancersResponse{
+				LoadBalancers: &[]loadbalancer.LoadBalancer{
+					*fixtureLoadBalancer(),
+				},
+			},
+			expectedCredentials: []loadbalancer.CredentialsResponse{
+				{
+					CredentialsRef: utils.Ptr("credentials-ref-3"),
+					DisplayName:    utils.Ptr("credentials-3"),
+					Username:       utils.Ptr("user-3"),
+				},
+			},
+			isValid: true,
+		},
+		{
+			description:         "no credentials",
+			allCredentials:      []loadbalancer.CredentialsResponse{},
+			expectedCredentials: []loadbalancer.CredentialsResponse{},
+			isValid:             true,
+		},
+		{
+			description:            "list load balancers fails",
+			listLoadBalancersFails: true,
+			isValid:                false,
+		},
```

Comment:
Don't you need to pass a `filterOp` to these tests? Otherwise the error will be because of the invalid filter, and not the actual error of the use case we are aiming to test